lib/net/dhcp/record.rb

```diff
@@ -10,8 +10,16 @@ def initialize(opts = { })
       self.ip      = Net::Validations.validate_ip! self.ip
     end
 
+    def lease?
+      type && type == "lease"
+    end
+
+    def reservation?
+      !lease?
```

Comment:
I think the 2nd option sounds niced from the user POV, the behavior of the previous implementation would still work. If you can do that and mark it as deprecated I think it'd be the best.